extensions/storages/source/providers/FileSystem.ts

```diff
@@ -1,46 +1,48 @@
 import { type Readable } from 'node:stream'
 import { dirname, join } from 'node:path'
-import { createReadStream } from 'node:fs'
 import fs from 'node:fs/promises'
-import fse from 'fs-extra'
+import { fileURLToPath } from 'node:url'
+import assert from 'node:assert'
 import { type Provider } from '../Provider'
 
 export class FileSystem implements Provider {
   protected readonly path: string
 
   public constructor (url: URL) {
-    if (url.host !== '')
-      throw new Error('File system URL must not contain host')
+    assert.equal(url.protocol, 'file:', `Invalid FileSystem URL: ${url.toString()}`)
 
-    this.path = url.pathname
+    this.path = fileURLToPath(url)
   }
 
   public async get (path: string): Promise<Readable | null> {
-    path = join(this.path, path)
+    try {
+      const fd = await fs.open(join(this.path, path))
 
-    if (!await fse.exists(path))
-      return null
+      return fd.createReadStream()
+    } catch (err) {
+      if ((err as NodeJS.ErrnoException)?.code === 'ENOENT') return null
 
-    return createReadStream(path)
+      throw err
+    }
   }
 
   public async put (rel: string, filename: string, stream: Readable): Promise<void> {
     const dir = join(this.path, rel)
     const path = join(dir, filename)
 
-    await fse.ensureDir(dir)
+    await fs.mkdir(dirname(path), { recursive: true })
```

Comment:
I think this is not equivalent. If given path (partially) exists, 'mkdir' will throw an exception. Or an I wrong?

Reply:
You are wrong, `recursive` will not throw